src/els_implementation_provider.erl

```diff
@@ -0,0 +1,100 @@
+-module(els_implementation_provider).
+
+-behaviour(els_provider).
+
+-include("erlang_ls.hrl").
+
+-export([ handle_request/2
+        , is_enabled/0
+        ]).
+
+%%==============================================================================
+%% els_provider functions
+%%==============================================================================
+
+-spec is_enabled() -> boolean().
+is_enabled() ->
+  true.
+
+-spec handle_request(tuple(), els_provider:state()) ->
+   {null | location(), els_provider:state()}.
+handle_request({implementation, Params}, State) ->
+  #{ <<"position">>     := #{ <<"line">>      := Line
+                            , <<"character">> := Character
+                            }
+   , <<"textDocument">> := #{<<"uri">> := Uri}
+   } = Params,
+  {ok, Document} = els_utils:lookup_document(Uri),
+  case find_implementation(Document, Line, Character) of
+    [#{range := Range}|_] ->
+      {#{uri => Uri, range => els_protocol:range(Range)}, State};
+    [] ->
+      {null, State}
+  end.
+
+%%==============================================================================
+%% Internal functions
+%%==============================================================================
+-spec find_implementation( els_dt_document:item()
+                         , non_neg_integer()
+                         , non_neg_integer()
+                         ) -> [poi()].
+find_implementation(Document, Line, Character) ->
+  case els_dt_document:get_element_at_pos(Document, Line + 1, Character + 1) of
+    [POI|_] -> implementation(Document, POI);
+    []      -> []
+  end.
+
+-spec implementation(els_dt_document:item(), poi()) -> [poi()].
+implementation(Document, #{kind := application, id := MFA}) ->
+  case callback(MFA) of
+    {CF, CA} ->
+      POIs = els_dt_document:pois(Document, [function]),
+      [POI || #{id := {F, A}} = POI <- POIs, F =:= CF, A =:= CA];
+    undefined ->
+      []
+  end;
+implementation(_Document, _POI) ->
+  [].
+
+-spec callback(mfa()) -> {atom(), non_neg_integer()} | undefined.
+%% gen_event
+callback({gen_event, add_handler, 3})     -> {init, 1};
```

Comment:
😱

Reply:
Is the 😱because of the manually maintained list? That list comes from the OTP documentation and it only changes sporadically.

Reply:
Yeah, that was what the face was about 😜.